### PR TITLE
Handle super-interfaces when retrieving lambda FI methods.

### PR DESCRIPTION
**Pull request by lazaroclapp**
This addresses #16.

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java [hunk 1]
```diff
@@ -62,27 +65,45 @@ private NullabilityUtil() { }
...
         // we want the method symbol for the single function inside the interface...hrm
-        List<Symbol> enclosedElements = funcInterfaceType.tsym.getEnclosedElements();
+        List<Symbol> enclosedElements;
 
         Symbol.MethodSymbol result = null;
-        for (Symbol s: enclosedElements) {
-            Symbol.MethodSymbol elem = (Symbol.MethodSymbol) s;
-            if (elem.isDefault() || elem.isStatic()) {
-                continue;
+        do {
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java]: Can we clean this up by passing in a `com.sun.tools.javac.code.Types` object and then using `Types.closure()`?  E.g. see [this code](https://github.com/google/error-prone/blob/master/check_api/src/main/java/com/google/errorprone/util/ASTHelpers.java#L503).  That way we don't need to explicitly traverse to supertypes, check for duplicates, etc.

Comment (lazaroclapp) [reply nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java]: Sounds good to me. Fixed now :)

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java [hunk 2]
```diff
@@ -60,31 +62,37 @@ private NullabilityUtil() { }
      * @param tree the lambda expression
      * @return the functional interface method
      */
-    public static Symbol.MethodSymbol getFunctionalInterfaceMethod(LambdaExpressionTree tree) {
+    public static Symbol.MethodSymbol getFunctionalInterfaceMethod(LambdaExpressionTree tree, Types types) {
         Type funcInterfaceType = ((JCTree.JCLambda) tree).type;
         // we want the method symbol for the single function inside the interface...hrm
-        List<Symbol> enclosedElements = funcInterfaceType.tsym.getEnclosedElements();
+        List<Symbol> enclosedElements;
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java]: Declare inside the loop?

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java [hunk 3]
```diff
@@ -60,31 +62,37 @@ private NullabilityUtil() { }
...
 
         Symbol.MethodSymbol result = null;
-        for (Symbol s: enclosedElements) {
-            Symbol.MethodSymbol elem = (Symbol.MethodSymbol) s;
-            if (elem.isDefault() || elem.isStatic()) {
-                continue;
+        for (Type t : types.closure(funcInterfaceType)) {
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java]: Sweet, this is much cleaner

**Pull request merged**